### PR TITLE
go database/sql sqli rule

### PR DESCRIPTION
**Pull request by colleend**
Go SQLi rules. Includes support for:
* [x] database/sql driver
* [x] go-pg driver (popular ORM for postgresql)
* [x] pgx driver (popular for postgresql, replaced lib/pq)

### CODE REVIEW

File: go/lang/security/audit/sqli/gosql-sqli.go [hunk 1]
```diff
@@ -1,75 +1,72 @@
-package main
+ipackage main
```
Comment (minusworld) [inline go/lang/security/audit/sqli/gosql-sqli.go]: Is `ipackage` a typo?

Comment (minusworld) [reply go/lang/security/audit/sqli/gosql-sqli.go]: A `vim` oopsie maybe? lol

**Review by minusworld — COMMENTED** [review]
Do you think these rules would benefit from the Go typing in Semgrep? I'm not sure how robust it is currently. cc @emjin

File: go/lang/security/audit/sqli/gosql-sqli.go (hunk 1 above)
Comment (colleend) [reply go/lang/security/audit/sqli/gosql-sqli.go]: good catch!!! definitely a vim oopsie 😅

**Comment by colleend** [comment]
Good question!! I think these rules are good even without Go typing (as we're mainly trying to catch variations of string concatenation that goes into a SQL query). My thought is that constant propagation is more helpful in this case, as we can use it to catch scenarios where a literal string is first assigned to a variable and then concatenated to a user-controlled variable 🤔 .

**Comment by minusworld** [comment]
sounds good!

**Review by minusworld — APPROVED** [review]

**Pull request merged**